docs/source/process.rst

```diff
@@ -0,0 +1,570 @@
+    'Amrozi accused his brother, whom he called " the witness ", of deliberately destroying his evidence.',
+    "Yucaipa owned Dominick 's before selling the chain to Safeway in 1998 for $ 2.5 billion .",
+    'Yucaipa owned Dominick Stores before selling the chain to Safeway in 1998 for $ 2.5 billion.',
+    "Yucaipa owned Dominick's before selling the chain to Safeway in 1998 for $ 2.5 billion.",
+    'Yucaipa owned Dominick Pizza before selling the chain to Safeway in 1998 for $ 2.5 billion.']
+
+For each original sentence, RoBERTA augmented a random word with three alternatives. In the first sentence, the word ``distorting`` is augmented with ``withholding``, ``suppressing``, and ``destroying``.
+
+Process multiple splits
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Many datasets have splits that we can process simultaneously with :func:`datasets.Dataset.map`. You can tokenize the ``sentence1`` field in the train and test split by:
+
+    >>> from datasets import load_dataset
+    >>>
+    # load all the splits
+    >>> dataset = load_dataset('glue', 'mrpc')
+    >>> encoded_dataset = dataset.map(lambda examples: tokenizer(examples['sentence1']), batched=True)
+    >>> encoded_dataset["train"][0]
+    {'sentence1': 'Amrozi accused his brother , whom he called " the witness " , of deliberately distorting his evidence .',
+    'sentence2': 'Referring to him as only " the witness " , Amrozi accused his brother of deliberately distorting his evidence .',
+    'label': 1,
+    'idx': 0,
+    'input_ids': [  101,  7277,  2180,  5303,  4806,  1117,  1711,   117,  2292, 1119,  1270,   107,  1103,  7737,   107,   117,  1104,  9938, 4267, 12223, 21811,  1117,  2554,   119,   102],
+    'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+    }
+
+Distributed usage
+^^^^^^^^^^^^^^^^^
+
+When you use :func:`datasets.Dataset.map` in a distributed setting, you should also use `torch.distributed.barrier <https://pytorch.org/docs/stable/distributed.html?highlight=barrier#torch.distributed.barrier>`_. This ensures the main process perform the mapping, while the other processes load the results, thereby avoiding duplicate work. The following example shows how you can use ``torch.distributed.barrier`` to synchronize the processes:
+
+   >>> from datasets import Dataset
+   >>> import torch.distributed
+   >>>
+   >>> dataset1 = Dataset.from_dict({"a": [0, 1, 2]})
+   >>>
+   >>> if training_args.local_rank > 0:
+   ...     print("Waiting for main process to perform the mapping")
+   ...     torch.distributed.barrier()
+   >>>
+   >>> dataset2 = dataset1.map(lambda x: {"a": x["a"] + 1})
+   >>>
+   >>> if training_args.local_rank == 0:
+   ...     print("Loading results from main process")
+   ...     torch.distributed.barrier()
+
+:class:`IterableDataset`
```

Comment:
Maybe we can have a separate page about how to process an iterable dataset ? The paradigm is a bit different from the on of regular map-style `Dataset` objects

Reply:
If we create a separate page for processing an `IterableDataset`, what do you think about combining that with how-to stream a dataset (I think you were referencing that [here](https://github.com/huggingface/datasets/pull/2718#discussion_r679134995))? Even though these are kind of two separate types of operations (loading vs processing), I think you are right that an `IterableDataset` is unique enough to warrant its own page :)